File: 3753-maximum-difference-between-even-and-odd-frequency-i/3753-maximum-difference-between-even-and-odd-frequency-i.py
class Solution:
    def maxDifference(self, s: str) -> int:
        frequency = {}

        for char in s:
            frequency[char] = frequency.get(char, 0) + 1

        odd_freqs = [freq for freq in frequency.values() if freq % 2 == 1]
        even_freqs = [freq for freq in frequency.values() if freq % 2 == 0]

        if odd_freqs and even_freqs:
            max_diff = max(odd_freqs) - min(even_freqs)
            return max_diff
        else:
            return 0
        

File: 3753-maximum-difference-between-even-and-odd-frequency-i/3753-maximum-difference-between-even-and-odd-frequency-i.java
class Solution {
    public int maxDifference(String s) {
        Map<Character, Integer> frequency = new HashMap<>();

        for (char c : s.toCharArray()) {
            frequency.put(c, frequency.getOrDefault(c, 0) + 1);
        }

        List<Integer> oddFreqs = new ArrayList<>();
        List<Integer> evenFreqs = new ArrayList<>();

        for (int freq : frequency.values()) {
            if (freq % 2 == 1) {
                oddFreqs.add(freq);
            } else {
                evenFreqs.add(freq);
            }
        }

        if (!oddFreqs.isEmpty() && !evenFreqs.isEmpty()) {
            int maxOdd = Collections.max(oddFreqs);
            int minEven = Collections.min(evenFreqs);
            return maxOdd - minEven;
        } else {
            return 0;
        }
    }
}